hack/lib/golang.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+
+#
+# Copyright 2023. The Bestchains Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+function go::build_binary() {
+	local -r targets=($@)
+
+	IFS="," read -ra platforms <<<"${BUILD_PLATFORMS-}"
+	if [[ ${#platforms[@]} -eq 0 ]]; then
+		read -ra platforms <<<$(go::host_platform)
+	fi
+
+	echo "Go version: $(go version)"
+
+	LDFLAGS+=" $(version::ldflags)"
+
+	for platform in "${platforms[@]}"; do
+		for target in "${targets[@]}"; do
+			echo "Start building ${target} for ${platform}:"
+			go::build_binary_for_platform "${target}" "${platform}"
+		done
+	done
+}
+
+function go::build_binary_for_platform() {
+	local -r target=$1
+	local -r platform=$2
+	local -r os=${platform%/*}
+	local -r arch=${platform##*/}
+
+	set -x
+	GO111MODULE=on CGO_ENABLED=0 GOOS=${os} GOARCH=${arch} go build -v \
+		-ldflags "${LDFLAGS-}" \
+		-o "_output/bin/${platform}/$target" \
+		"cmd/$target/main.go"
+	set +x
+}
+
+function go::host_platform() {
+	echo "$(go env GOHOSTOS)/$(go env GOHOSTARCH)"
+}
+
+function go::verify_go_version() {
+	if [[ -z "$(command -v go)" ]]; then
+		echo "go is not installed"
+		exit 1
+	fi
+
+	local go_version
+	IFS=" " read -ra go_version <<<"$(GOFLAGS='' go version)"
+	local minimum_go_version
+	minimum_go_version=go1.18.3
```

Comment:
when go.mod set go version is 1.20, minimum_go_version should also update to 1.20